dev/io.openliberty.http.monitor/src/io/openliberty/http/monitor/HttpStatAttributes.java

```diff
@@ -150,6 +192,121 @@ public String toString() {
 		+ "Response Status (Optional: can be -1): [%d] \n" 
 		+ "Error Type(Optional - can be empty): [%s]",
 				requestMethod, scheme, networkProtocolName, networkProtocolVersion, serverName, serverPort,
-				httpRoute.orElse(""), responseStatus.orElse(-1), errorType.orElse(""));
+				httpRoute, responseStatus, errorType);
+	}
+	
+	public static Builder builder() {
+		return new Builder();
+	}
+	
+	
+	public static class Builder {
+
+		private String requestMethod, scheme, networkProtocolName, networkProtocolVersion, serverName;
+		private int serverPort;
+
+		/*
+		 * Exception related fields are optional. We are unable to facilitate capturing
+		 * Exceptions But we will leave it here. Additional Context : We can capture
+		 * exceptions thrown by servlets by surrounding the the chainFilter with try
+		 * catch. But we have no way of capturing application exception of
+		 * Jaxrs/restfulws exceptions
+		 */
+		private Optional<Exception> exception = Optional.ofNullable(null);
+		private Optional<String> errorType = Optional.ofNullable(null);
+
+		/*
+		 * Conditionally required as per HTTP Semantics Convention
+		 */
+		private Optional<String> httpRoute = Optional.ofNullable(null);
+		private Optional<Integer> responseStatus = Optional.ofNullable(null);
+
+		/**
+		 * Builds an instance of {@link HttpStatAttributes} with values from this
+		 * builder. Will validate and throw an {@link IllegalStateException} if the
+		 * required fields are not filled.
+		 * 
+		 * @return Instance of {@link HttpStatAttributes}
+		 * @throws IllegalStateException
+		 */
+		@FFDCIgnore(value = { IllegalStateException.class })
+		public HttpStatAttributes build() {
+			if (!validate()) {
+				if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+					Tr.debug(tc, String.format("Invalid HTTP Stats attributes : \n %s", toString()));
+				}
+			}
+			
+			/*
+			 * Because of above check, we should never really actually come into this. 
+			 * The constructor does the same check again (for any calls made by others who somehow got their hands
+			 * on to an instance of a Builder.
+			 */
+			try {
+				return new HttpStatAttributes(this);
+			} catch(IllegalStateException ise) {
+				//do nothing
+			}
```

Comment:
```suggestion
			try {
				return new HttpStatAttributes(this);
			} catch (IllegalStateException ise) {
				//do nothing
				if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
					Tr.debug(tc, String.format("Invalid HTTP Stats attributes : \n %s", toString()));
				}
			}
```
Not sure why should validate twice.